src/astro/files/base.py

```diff
@@ -72,12 +73,21 @@ def create_from_dataframe(self, df: pd.DataFrame) -> None:
             self.type.create_from_dataframe(stream=stream, df=df)
 
     def export_to_dataframe(self, **kwargs) -> pd.DataFrame:
-        """Read file from all supported location and convert them into dataframes"""
+        """Read file from all supported location and convert them into dataframes.
+
+        Due to noted issues with using smart_open with pandas (like
+        https://github.com/RaRe-Technologies/smart_open/issues/524), we create a BytesIO or StringIO buffer
+        before exporting to a dataframe. We've found a sizable speed improvement with this optimization.
+        """
+
         mode = "rb" if self.is_binary() else "r"
```

Comment:
@dimberman - maybe we can move this to a function of it's own, might be useful at other locations.

Reply:
@utkarsharma2 good call. For now I've moved it into a private function, so if later someone has a different use for it they can make it public.